### ISSUE TITLE
MTK doesn't precompile on my Windows 11 laptop anymore

### ISSUE THREAD

**Issue by B-LIE** [SciML/ModelingToolkit.jl#2730]
**Describe the bug 🐞**

ModelingToolkit doesn't precompile on my Windows 11 laptop any more. [1TB free SSD space, 32 GB RAM, 12th gen i7 processor.]
* I have uninstalled Julia + deleted .julia + checked %appdata% for Julia info (couldn't find, but may have missed), restarted computer, and re-installed Julia v10.3
* I then *started* by adding `ModelingToolkit`; no other packages in the computer. I have tried with the latest version of MTK, without success. So now I'm trying with MTK v9.0.0, which definitely used to work in the past.
* Pkg manager seems to install/precompile a lot of dependencies. At the last step, 227/228, precompilation hangs for hours (8-10 hours), with no further progress.

**Expected behavior**

Obviously, the expected behavior is that precompile finishes in a reasonable time. I don't know what that reasonable time should be... perhaps some 10-20 minutes?? But definitely not hours upon hours.

**Minimal Reproducible Example 👇**

I don't really know what an MRE should be in this case... Obviously, something has messed up my computer... I don't even know where to start.

```julia

```

**Error & Stacktrace ⚠️**
After waiting for "eternity" without completion of precompilation, I Ctrl-C Julia, upon which the precompilation stops with a very long message. I then do a status check:

```julia
(@v1.10) pkg> status
Status `C:\Users\Bernt\.julia\environments\v1.10\Project.toml`
⌃ [961ee093] ModelingToolkit v9.0.0
Info Packages marked with ⌃ have new versions available and may be upgradable.
```
This may seem like somehow ModelingToolkit *was* precompiled, but if I go back to Julia and issue `using ModelingToolkit`, precompilation of the last step starts again...
<img width="632" alt="image" src="https://github.com/SciML/ModelingToolkit.jl/assets/54220415/1409a0c6-856d-4adc-8aff-91280d280438">

**Environment (please complete the following information):**

  - Output of `using Pkg; Pkg.status()`
As above:

```julia
(@v1.10) pkg> status
Status `C:\Users\Bernt\.julia\environments\v1.10\Project.toml`
⌃ [961ee093] ModelingToolkit v9.0.0
Info Packages marked with ⌃ have new versions available and may be upgradable.
```

  - Output of `using Pkg; Pkg.status(; mode = PKGMODE_MANIFEST)`

```julia
julia> using Pkg; Pkg.status(; mode = PKGMODE_MANIFEST)
Status `C:\Users\Bernt\.julia\environments\v1.10\Manifest.toml`
  [47edcb42] ADTypes v1.2.1
  [1520ce14] AbstractTrees v0.4.5
  [7d9f7c33] Accessors v0.1.36
  [79e6a3ab] Adapt v4.0.4
  [66dad0bd] AliasTables v1.1.3
  [ec485272] ArnoldiMethod v0.4.0
  [4fba245c] ArrayInterface v7.10.0
  [4c555306] ArrayLayouts v1.9.3
  [e2ed5e7c] Bijections v0.1.6
  [62783981] BitTwiddlingConvenienceFunctions v0.1.5
  [2a0fbf3d] CPUSummary v0.2.5
  [00ebfdb7] CSTParser v3.4.3
  [49dc2e85] Calculus v0.5.1
  [d360d2e6] ChainRulesCore v1.23.0
  [fb6a15b2] CloseOpenIntervals v0.1.12
  [861a8166] Combinatorics v1.0.2
  [a80b9123] CommonMark v0.8.12
  [38540f10] CommonSolve v0.2.4
  [bbf7d656] CommonSubexpressions v0.3.0
  [34da2185] Compat v4.15.0
  [b152e2b5] CompositeTypes v0.1.4
  [a33af91c] CompositionsBase v0.1.2
  [2569d6c7] ConcreteStructs v0.2.3
  [187b0558] ConstructionBase v1.5.5
  [adafc99b] CpuId v0.3.1
  [a8cc5b0e] Crayons v4.1.1
  [9a962f9c] DataAPI v1.16.0
  [864edb3b] DataStructures v0.18.20
  [e2d170a0] DataValueInterfaces v1.0.0
  [2b5f629d] DiffEqBase v6.151.1
⌅ [459566f4] DiffEqCallbacks v2.37.0
  [163ba53b] DiffResults v1.1.0
  [b552c78f] DiffRules v1.15.1
  [b4f34e82] Distances v0.10.11
  [31c24e10] Distributions v0.25.108
  [ffbed154] DocStringExtensions v0.9.3
  [5b8099bc] DomainSets v0.7.14
  [fa6b7ba4] DualNumbers v0.6.8
  [7c1d4256] DynamicPolynomials v0.5.7
⌅ [06fc5a27] DynamicQuantities v0.11.3
  [4e289a0a] EnumX v1.0.4
  [f151be2c] EnzymeCore v0.7.2
  [d4d017d3] ExponentialUtilities v1.26.1
  [e2ba6199] ExprTools v0.1.10
  [7034ab61] FastBroadcast v0.2.8
  [9aa1b823] FastClosures v0.3.2
  [29a986be] FastLapackInterface v2.0.4
  [1a297f60] FillArrays v1.11.0
  [64ca27bc] FindFirstFunctions v1.2.0
  [6a86dc24] FiniteDiff v2.23.1
  [1fa38f19] Format v1.3.7
  [f6369f11] ForwardDiff v0.10.36
  [069b7b12] FunctionWrappers v1.1.3
  [77dc65aa] FunctionWrappersWrappers v0.1.3
  [d9f16b24] Functors v0.4.10
  [46192b85] GPUArraysCore v0.1.6
  [c145ed77] GenericSchur v0.5.4
  [c27321d9] Glob v1.3.1
  [86223c79] Graphs v1.11.0
  [3e5b6fbb] HostCPUFeatures v0.1.16
  [34004b35] HypergeometricFunctions v0.3.23
  [615f187c] IfElse v0.1.1
  [d25df0c9] Inflate v0.1.4
  [8197267c] IntervalSets v0.7.10
  [3587e190] InverseFunctions v0.1.14
  [92d709cd] IrrationalConstants v0.2.2
  [82899510] IteratorInterfaceExtensions v1.0.0
  [692b3bcd] JLLWrappers v1.5.0
  [682c06a0] JSON v0.21.4
  [98e50ef6] JuliaFormatter v1.0.56
  [ccbc3e58] JumpProcesses v9.11.1
  [ef3ab10e] KLU v0.6.0
  [ba0b0d4f] Krylov v0.9.6
  [b964fa9f] LaTeXStrings v1.3.1
  [2ee39098] LabelledArrays v1.16.0
  [984bce1d] LambertW v0.4.6
  [23fbe1c1] Latexify v0.16.3
  [10f19ff3] LayoutPointers v0.1.15
⌅ [5078a376] LazyArrays v1.10.0
  [d3d80556] LineSearches v7.2.0
  [7ed4a6bd] LinearSolve v2.30.0
  [2ab3a3ac] LogExpFunctions v0.3.27
  [bdcacae8] LoopVectorization v0.12.170
  [d8e11817] MLStyle v0.4.17
  [1914dd2f] MacroTools v0.5.13
  [d125e4d3] ManualMemory v0.1.8
⌅ [a3b82374] MatrixFactorizations v2.2.0
  [bb5d69b7] MaybeInplace v0.1.2
  [e1d29d7a] Missings v1.2.0
⌃ [961ee093] ModelingToolkit v9.0.0
  [46d2c3a1] MuladdMacro v0.2.4
  [102ac46a] MultivariatePolynomials v0.5.5
  [d8a4904e] MutableArithmetics v1.4.4
  [d41bc354] NLSolversBase v7.8.3
  [2774e3e8] NLsolve v4.5.1
  [77ba4419] NaNMath v1.0.2
  [8913a72c] NonlinearSolve v3.12.0
  [6fe1bfb0] OffsetArrays v1.14.0
  [bac558e1] OrderedCollections v1.6.3
  [1dea7af3] OrdinaryDiffEq v6.78.0
  [90014a1f] PDMats v0.11.31
  [65ce6f38] PackageExtensionCompat v1.0.2
  [d96e819e] Parameters v0.12.3
  [69de0a69] Parsers v2.8.1
  [e409e4f3] PoissonRandom v0.4.4
  [f517fe37] Polyester v0.7.14
  [1d0040c9] PolyesterWeave v0.2.1
  [d236fae5] PreallocationTools v0.4.21
  [aea7be01] PrecompileTools v1.2.1
  [21216c6a] Preferences v1.4.3
  [43287f4e] PtrArrays v1.1.0
  [1fd47b50] QuadGK v2.9.4
  [e6cf234a] RandomNumbers v1.5.3
  [3cdcf5f2] RecipesBase v1.3.4
  [731186ca] RecursiveArrayTools v3.19.0
  [f2c3362d] RecursiveFactorization v0.2.23
  [189a3867] Reexport v1.2.2
  [ae029012] Requires v1.3.0
  [79098fc4] Rmath v0.7.1
  [7e49a35a] RuntimeGeneratedFunctions v0.5.13
  [94e857df] SIMDTypes v0.1.0
  [476501e8] SLEEFPirates v0.6.42
  [0bca4576] SciMLBase v2.38.0
  [c0aeaf25] SciMLOperators v0.3.8
  [53ae85a6] SciMLStructures v1.2.0
  [efcf1570] Setfield v1.1.1
  [727e6d20] SimpleNonlinearSolve v1.8.0
  [699a6c99] SimpleTraits v0.9.4
  [ce78b400] SimpleUnPack v1.1.0
  [a2af1166] SortingAlgorithms v1.2.1
  [47a9eef4] SparseDiffTools v2.19.0
  [e56a9233] Sparspak v0.3.9
  [276daf66] SpecialFunctions v2.4.0
  [aedffcd0] Static v0.8.10
  [0d7ed370] StaticArrayInterface v1.5.0
  [90137ffa] StaticArrays v1.9.4
  [1e83bf80] StaticArraysCore v1.4.2
  [82ae8749] StatsAPI v1.7.0
  [2913bbd2] StatsBase v0.34.3
  [4c63d2b9] StatsFuns v1.3.1
  [7792a7ef] StrideArraysCore v0.5.6
  [2efcf032] SymbolicIndexingInterface v0.3.21
  [19f23fe9] SymbolicLimits v0.2.0
  [d1185830] SymbolicUtils v1.6.0
  [0c5d862f] Symbolics v5.28.0
  [3783bdb8] TableTraits v1.0.1
  [bd369af6] Tables v1.11.1
  [8ea1fca8] TermInterface v0.4.1
  [8290d209] ThreadingUtilities v0.5.2
  [a759f4b9] TimerOutputs v0.5.24
  [0796e94c] Tokenize v0.5.29
  [d5829a12] TriangularSolve v0.2.0
  [410a4b4d] Tricks v0.1.8
  [781d530d] TruncatedStacktraces v1.4.0
  [5c2747f8] URIs v1.5.1
  [3a884ed6] UnPack v1.0.2
  [1986cc42] Unitful v1.20.0
  [a7c27f48] Unityper v0.1.6
  [3d5dd08c] VectorizationBase v0.21.67
  [19fa3120] VertexSafeGraphs v0.2.0
  [1d5cc7b8] IntelOpenMP_jll v2024.1.0+0
  [856f044c] MKL_jll v2024.1.0+0
  [efe28fd5] OpenSpecFun_jll v0.5.5+0
  [f50d1b31] Rmath_jll v0.4.2+0
  [1317d2d5] oneTBB_jll v2021.12.0+0
  [0dad84c5] ArgTools v1.1.1
  [56f22d72] Artifacts
  [2a0f44e3] Base64
  [ade2ca70] Dates
  [8ba89e20] Distributed
  [f43a241f] Downloads v1.6.0
  [7b1f6079] FileWatching
  [9fa8497b] Future
  [b77e0a4c] InteractiveUtils
  [4af54fe1] LazyArtifacts
  [b27032c2] LibCURL v0.6.4
  [76f85450] LibGit2
  [8f399da3] Libdl
  [37e2e46d] LinearAlgebra
  [56ddb016] Logging
  [d6f4376e] Markdown
  [a63ad114] Mmap
  [ca575930] NetworkOptions v1.2.0
  [44cfe95a] Pkg v1.10.0
  [de0858da] Printf
  [3fa0cd96] REPL
  [9a3f8284] Random
  [ea8e919c] SHA v0.7.0
  [9e88b42a] Serialization
  [1a1011a3] SharedArrays
  [6462fe0b] Sockets
  [2f01184e] SparseArrays v1.10.0
  [10745b16] Statistics v1.10.0
  [4607b0f0] SuiteSparse
  [fa267f1f] TOML v1.0.3
  [a4e569a6] Tar v1.10.0
  [8dfed614] Test
  [cf7118a7] UUIDs
  [4ec0a83e] Unicode
  [e66e0078] CompilerSupportLibraries_jll v1.1.1+0
  [deac9b47] LibCURL_jll v8.4.0+0
  [e37daf67] LibGit2_jll v1.6.4+0
  [29816b5a] LibSSH2_jll v1.11.0+1
  [c8ffd9c3] MbedTLS_jll v2.28.2+1
  [14a3606d] MozillaCACerts_jll v2023.1.10
  [4536629a] OpenBLAS_jll v0.3.23+4
  [05823500] OpenLibm_jll v0.8.1+2
  [bea87d4a] SuiteSparse_jll v7.2.1+1
  [83775a58] Zlib_jll v1.2.13+1
  [8e850b90] libblastrampoline_jll v5.8.0+1
  [8e850ede] nghttp2_jll v1.52.0+1
  [3f19e933] p7zip_jll v17.4.0+2
Info Packages marked with ⌃ and ⌅ have new versions available. Those with ⌃ may be upgradable, but those with ⌅ are restricted by compatibility constraints from upgrading. To see why use `status --outdated -m`
```

  - Output of `versioninfo()`

```julia
 versioninfo()
Julia Version 1.10.3
Commit 0b4590a550 (2024-04-30 10:59 UTC)
Build Info:
  Official https://julialang.org/ release
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: 12 × 12th Gen Intel(R) Core(TM) i7-1255U
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-15.0.7 (ORCJIT, alderlake)
Threads: 1 default, 0 interactive, 1 GC (on 12 virtual cores)
```

**Additional context**

A few days ago, I installed some new packages on my Julia set-up [AbstractAlgebra, Nemo, Groebner. I tried to install Oscar, but that didn't precompile and told me it doesn't work under Windows -- needs Linux subsystem for Windows, which I don't want to install. I also installed some Python packages, such as SymPy.] Subsequently, I removed these Julia packages. But now ModelingToolkit [MTK] doesn't precompile any more. Here is what I've tried to do:
* I removed all packages in Julia. Then I restarted computer, added MTK. It starts to precompile, but hangs for 12 h+ at the last step.
* I then uninstalled Julia, deleted .julia, restarted computer, re-installed Julia, and added MTK. Same as before. Stops on the last step (222/223) and hangs for 8+ hours.
* I'm have now started to install *old* versions of MTK that definitely worked in the past. Currently, I try to install MTK v.9.0.0. It hangs at the last step: 227/228 for that particular version.

**Comment by finmod** [comment]
Confirm the same problem that started three days ago. On a completely clean install of Julia 1.10.3 via juliaup for windows, the error message is:

    _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.10.3 (2024-04-30)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

(@v1.10) pkg> up
    Updating registry at `C:\Users\Denis\.julia\registries\General.toml`
   Installed FastLapackInterface ─ v2.0.4
  No Changes to `C:\Users\Denis\.julia\environments\v1.10\Project.toml`
    Updating `C:\Users\Denis\.julia\environments\v1.10\Manifest.toml`
  [29a986be] ↑ FastLapackInterface v2.0.3 ⇒ v2.0.4
Precompiling project...
┌ Warning: attempting to remove probably stale pidfile  9/15
│   path = "C:\\Users\\Denis\\.julia\\compiled\\v1.10\\ModelingToolkit\\Qmdqu_BUi1j.ji.pidfile"
└ @ FileWatching.Pidfile C:\Users\Denis\.julia\juliaup\julia-1.10.3+0.x64.w64.mingw32\share\julia\stdlib\v1.10\FileWatching\src\pidfile.jl:244
  Progress [======================================>  ]  14/15
  ◐ ModelingToolkit

**Comment by SA8416** [comment]
I'm also having this issue, on a fresh install of windows 11 and of julia 1.10.3. 

I get a similar output, with pre-compilation never completing:

```
Precompiling ModelingToolkit
┌ Warning: attempting to remove probably stale pidfile  1/3
│   path = "C:\\Users\\user\\.julia\\compiled\\v1.10\\ModelingToolkit\\Qmdqu_HW5VR.ji.pidfile"
└ @ FileWatching.Pidfile C:\Users\user\.julia\juliaup\julia-1.10.3+0.x64.w64.mingw32\share\julia\stdlib\v1.10\FileWatching\src\pidfile.jl:244
        Info Given ModelingToolkit was explicitly requested, output will be shown live

Please submit a bug report with steps to reproduce this fault, and any error messages that follow (in their entirety). Thanks.
Exception: EXCEPTION_ACCESS_VIOLATION at 0x7ffc6bf5845f -- gc_setmark_pool_ at C:/workdir/src\gc.c:876 [inlined]
gc_setmark_pool at C:/workdir/src\gc.c:895 [inlined]
gc_setmark at C:/workdir/src\gc.c:902 [inlined]
gc_setmark at C:/workdir/src\gc.c:898 [inlined]
gc_mark_outrefs at C:/workdir/src\gc.c:2617 [inlined]
gc_mark_loop_serial_ at C:/workdir/src\gc.c:2690
in expression starting at C:\Users\user\.julia\packages\ModelingToolkit\Gpzyo\src\ModelingToolkit.jl:6
gc_setmark_pool_ at C:/workdir/src\gc.c:876 [inlined]
gc_setmark_pool at C:/workdir/src\gc.c:895 [inlined]
gc_setmark at C:/workdir/src\gc.c:902 [inlined]
gc_setmark at C:/workdir/src\gc.c:898 [inlined]
gc_mark_outrefs at C:/workdir/src\gc.c:2617 [inlined]
gc_mark_loop_serial_ at C:/workdir/src\gc.c:2690
gc_mark_loop_serial at C:/workdir/src\gc.c:2713
gc_mark_loop at C:/workdir/src\gc.c:2894 [inlined]
_jl_gc_collect at C:/workdir/src\gc.c:3227
ijl_gc_collect at C:/workdir/src\gc.c:3524
maybe_collect at C:/workdir/src\gc.c:937 [inlined]
jl_gc_pool_alloc_inner at C:/workdir/src\gc.c:1293 [inlined]
ijl_gc_pool_alloc at C:/workdir/src\gc.c:1341
argextype at .\compiler\optimize.jl:344                                             
argextype at .\compiler\optimize.jl:341                                             
argextype at .\compiler\optimize.jl:340 [inlined]
is_known_call at .\compiler/ssair\passes.jl:5
sroa_pass! at .\compiler/ssair\passes.jl:1032
run_passes at .\compiler\optimize.jl:489
run_passes at .\compiler\optimize.jl:504 [inlined]
optimize at .\compiler\optimize.jl:453 [inlined]
_typeinf at .\compiler\typeinfer.jl:272
typeinf at .\compiler\typeinfer.jl:216
typeinf_edge at .\compiler\typeinfer.jl:930
abstract_call_method at .\compiler\abstractinterpretation.jl:629
abstract_call_gf_by_type at .\compiler\abstractinterpretation.jl:95
abstract_call_known at .\compiler\abstractinterpretation.jl:2087
abstract_call at .\compiler\abstractinterpretation.jl:2169
abstract_apply at .\compiler\abstractinterpretation.jl:1612
abstract_call_known at .\compiler\abstractinterpretation.jl:2004
abstract_call at .\compiler\abstractinterpretation.jl:2169
abstract_call at .\compiler\abstractinterpretation.jl:2162
abstract_call at .\compiler\abstractinterpretation.jl:2354
abstract_eval_call at .\compiler\abstractinterpretation.jl:2370
abstract_eval_statement_expr at .\compiler\abstractinterpretation.jl:2380
abstract_eval_statement at .\compiler\abstractinterpretation.jl:2624
abstract_eval_basic_statement at .\compiler\abstractinterpretation.jl:2913
typeinf_local at .\compiler\abstractinterpretation.jl:3098
typeinf_nocycle at .\compiler\abstractinterpretation.jl:3186
_typeinf at .\compiler\typeinfer.jl:247
typeinf at .\compiler\typeinfer.jl:216
typeinf_edge at .\compiler\typeinfer.jl:930
abstract_call_method at .\compiler\abstractinterpretation.jl:629
abstract_call_gf_by_type at .\compiler\abstractinterpretation.jl:95
abstract_call_known at .\compiler\abstractinterpretation.jl:2087
abstract_call at .\compiler\abstractinterpretation.jl:2169
abstract_call at .\compiler\abstractinterpretation.jl:2162
abstract_call at .\compiler\abstractinterpretation.jl:2354
abstract_eval_call at .\compiler\abstractinterpretation.jl:2370
abstract_eval_statement_expr at .\compiler\abstractinterpretation.jl:2380
abstract_eval_statement at .\compiler\abstractinterpretation.jl:2624
abstract_eval_basic_statement at .\compiler\abstractinterpretation.jl:2913
typeinf_local at .\compiler\abstractinterpretation.jl:3098
typeinf_nocycle at .\compiler\abstractinterpretation.jl:3186
_typeinf at .\compiler\typeinfer.jl:247
typeinf at .\compiler\typeinfer.jl:216
typeinf_ext at .\compiler\typeinfer.jl:1051
typeinf_ext_toplevel at .\compiler\typeinfer.jl:1082
typeinf_ext_toplevel at .\compiler\typeinfer.jl:1078
jfptr_typeinf_ext_toplevel_38978.1 at C:\Users\user\.julia\juliaup\julia-1.10.3+0.x64.w64.mingw32\lib\julia\sys.dll (unknown line)
_jl_invoke at C:/workdir/src\gf.c:2895 [inlined]
ijl_apply_generic at C:/workdir/src\gf.c:3077 [inlined]
jl_apply at C:/workdir/src\julia.h:1982 [inlined]
jl_type_infer at C:/workdir/src\gf.c:394
_generate_from_hint at C:/workdir/src\gf.c:2756 [inlined]
jl_compile_now at C:/workdir/src\gf.c:2770 [inlined]
ijl_compile_method_instance at C:/workdir/src\gf.c:2782
ijl_compile_hint at C:/workdir/src\gf.c:2820
precompile at .\loading.jl:3210
precompile_mi at C:\Users\user\.julia\packages\PrecompileTools\L8A3n\src\PrecompileTools.jl:13
foreach at .\abstractarray.jl:3097
recompile_invalidations at C:\Users\user\.julia\packages\PrecompileTools\L8A3n\src\invalidations.jl:23
unknown function (ip: 000001f367eb1973)
jl_apply at C:/workdir/src\julia.h:1982 [inlined]
do_call at C:/workdir/src\interpreter.c:126
eval_value at C:/workdir/src\interpreter.c:223
eval_stmt_value at C:/workdir/src\interpreter.c:174 [inlined]
eval_body at C:/workdir/src\interpreter.c:635
jl_interpret_toplevel_thunk at C:/workdir/src\interpreter.c:775
jl_toplevel_eval_flex at C:/workdir/src\toplevel.c:934
jl_eval_module_expr at C:/workdir/src\toplevel.c:215 [inlined]
jl_toplevel_eval_flex at C:/workdir/src\toplevel.c:736
jl_toplevel_eval_flex at C:/workdir/src\toplevel.c:877
jl_toplevel_eval_flex at C:/workdir/src\toplevel.c:877
ijl_toplevel_eval at C:/workdir/src\toplevel.c:943 [inlined]
ijl_toplevel_eval_in at C:/workdir/src\toplevel.c:985
eval at .\boot.jl:385 [inlined]
include_string at .\loading.jl:2076
_include at .\loading.jl:2136
include at .\Base.jl:495 [inlined]
include_package_for_output at .\loading.jl:2222
jfptr_include_package_for_output_84841.1 at C:\Users\user\.julia\juliaup\julia-1.10.3+0.x64.w64.mingw32\lib\julia\sys.dll (unknown line)
jl_apply at C:/workdir/src\julia.h:1982 [inlined]
do_call at C:/workdir/src\interpreter.c:126
eval_value at C:/workdir/src\interpreter.c:223
eval_stmt_value at C:/workdir/src\interpreter.c:174 [inlined]
eval_body at C:/workdir/src\interpreter.c:635
jl_interpret_toplevel_thunk at C:/workdir/src\interpreter.c:775
jl_toplevel_eval_flex at C:/workdir/src\toplevel.c:934
jl_toplevel_eval_flex at C:/workdir/src\toplevel.c:877
ijl_toplevel_eval at C:/workdir/src\toplevel.c:943 [inlined]
ijl_toplevel_eval_in at C:/workdir/src\toplevel.c:985
eval at .\boot.jl:385 [inlined]
include_string at .\loading.jl:2076
include_string at .\loading.jl:2086 [inlined]
exec_options at .\client.jl:316
_start at .\client.jl:552
jfptr__start_86582.1 at C:\Users\user\.julia\juliaup\julia-1.10.3+0.x64.w64.mingw32\lib\julia\sys.dll (unknown line)
jl_apply at C:/workdir/src\julia.h:1982 [inlined]
true_main at C:/workdir/src\jlapi.c:582
jl_repl_entrypoint at C:/workdir/src\jlapi.c:731
mainCRTStartup at C:/workdir/cli\loader_exe.c:58
BaseThreadInitThunk at C:\Windows\System32\KERNEL32.DLL (unknown line)
RtlUserThreadStart at C:\Windows\SYSTEM32\ntdll.dll (unknown line)
Allocations: 13613850 (Pool: 13597138; Big: 16712); GC: 14
Progress [====================>                    ]  1/2
  ◓ ModelingToolkit
```

**Comment by B-LIE** [comment]
I have now tried with Julia v1.10.3 and latest version of MTK + MTK v9.13.0 and MTK v9.0.0, as well as Julia v1.10.2 and the same trio of MTK versions and Julia v1.10.0 and the same MTK version trios. No luck.

**Comment by skycolt** [comment]
same problem. Even I don't add ModelingToolkit.jl explicitly, once it's required the precompile will halt here. 1.10.3.
I did try julia 1.9.4 and everything seems to still play well

**Comment by ChrisRackauckas** [comment]
We know, thread is here:

https://discourse.julialang.org/t/modelingtoolkit-takes-forever-to-precompile-on-windows-11/114492/44

**Comment by natlampen** [comment]
I believe this is the same as #2722?

**Comment by ChrisRackauckas** [comment]
Yes, and it's also the same as https://github.com/JuliaLang/julia/issues/54550, and it's also the same as https://discourse.julialang.org/t/modelingtoolkit-takes-forever-to-precompile-on-windows-11/114492 . We know it exists, it's all hands-on-deck madness, it's weird, its likely a GC bug in Julia, and we have no idea what's causing it but we're running around trying to get a solution ASAP. More threads will not help, we know the issue exists.

**Comment by ChrisRackauckas** [comment]
Because this was isolated to being a Julia issue, we can keep the conversation going there: https://github.com/JuliaLang/julia/issues/54550

**Comment by oscardssmith** [comment]
fixed for real in Julia [1.10.4](https://discourse.julialang.org/t/julia-v1-10-4-has-been-released/115197)
